Sources/SwiftFrameCore/Workers/ImageComposer.swift

```diff
@@ -0,0 +1,106 @@
+    func addTemplateImage(_ image: NSBitmapImageRep) throws {
+        guard let templateImage = image.cgImage else {
+            throw NSError(description: "Could not render template image")
+        }
+
+        context.saveGState()
+        defer { context.restoreGState() }
+
+        context.draw(templateImage, in: image.ky_nativeRect)
+    }
+
+    // MARK: - Titles Rendering
+
+    func addStrings(_ associatedStrings: [AssociatedString], maxFontSizeByGroup: [String: CGFloat], font: NSFont, color: NSColor, maxFontSize: CGFloat) throws {
+        try associatedStrings.forEach {
+            if let sharedSize = $0.data.groupIdentifier.flatMap({ maxFontSizeByGroup[$0] }) {
+                // Can use fixed font size since common maximum has already been calculated
+                try add(
+                    title: $0.string,
+                    font: $0.data.fontOverride?.font() ?? font,
+                    color: $0.data.textColorOverride ?? color,
+                    fixedFontSize: sharedSize,
+                    textData: $0.data)
+            } else {
+                try add(
+                    title: $0.string,
+                    font: $0.data.fontOverride?.font() ?? font,
+                    color: $0.data.textColorOverride ?? color,
+                    maxFontSize: $0.data.maxFontSizeOverride ?? maxFontSize,
+                    textData: $0.data)
+            }
+        }
+    }
+
+    @discardableResult private func add(title: String, font: NSFont, color: NSColor, maxFontSize: CGFloat, textData: TextData) throws -> CGFloat {
+        let fontSize = try textRenderer.maximumFontSizeThatFits(
+            string: title,
+            font: font,
+            alignment: textData.textAlignment,
+            maxSize: maxFontSize,
+            size: textData.rect.size)
+        let adaptedFont = font.ky_toFont(ofSize: fontSize)
+        try textRenderer.render(text: title, font: adaptedFont, color: color, alignment: textData.textAlignment, rect: textData.rect, context: context)
+        return fontSize
+    }
+
+    private func add(title: String, font: NSFont, color: NSColor, fixedFontSize: CGFloat, textData: TextData) throws {
+        let adaptedFont = font.ky_toFont(ofSize: fixedFontSize)
+        try textRenderer.render(text: title, font: adaptedFont, color: color, alignment: textData.textAlignment, rect: textData.rect, context: context)
+    }
+
+    // MARK: - Screenshots Rendering
+
+    func add(screenshots: [String: URL], with screenshotData: [ScreenshotData], for locale: String) throws {
+        try screenshotData.forEach { data in
+            guard let image = NSBitmapImageRep.ky_loadFromURL(screenshots[data.screenshotName]) else {
+                throw NSError(description: "Screenshot named \(data.screenshotName) not found in folder \"\(locale)\"")
+            }
+            try add(screenshot: image, with: data)
+        }
+    }
+
+    func add(screenshot: NSBitmapImageRep, with data: ScreenshotData) throws {
```

Comment:
Unless I missed something, this seems to only be used in the method above so could maybe also be inlined since it's just one line anyway.